src/trunk.c

```diff
@@ -7967,14 +7968,25 @@ trunk_print_pivots(platform_log_handle *log_handle,
         pivot_no++)
    {
       key pivot = trunk_get_pivot(spl, node, pivot_no);
+
       // Used to format generated string for negative / positive infinity keys.
       char buf[PIVOT_KEY_OUTBUF_LEN + 1];
-      snprintf(buf, sizeof(buf), "%-24s", key_string(spl->cfg.data_cfg, pivot));
+      memset(buf, ' ', sizeof(buf));
+      int nbytes = snprintf(buf,
+                            sizeof(buf),
+                            "%.*s",
+                            PIVOT_KEY_OUTBUF_LEN,
+                            key_string(spl->cfg.data_cfg, pivot));
+      if (nbytes <= PIVOT_KEY_OUTBUF_LEN) {
+         // Overwrite null-termination byte to blank-pad the rest of the buffer
+         buf[nbytes] = ' ';
```

Comment:
All this jugglery is needed because we want the pivot-key output to be exactly 24-bytes.

The `key_string()` method generates shorter-strings for some cases: `"(negative_infinity)"` and 
`"(positive_infinity)"`.

So, what's happening here is that when the `snprintf()`-ed string is shorter than the requested 24 bytes, we _know_ that it's one of these two shorter-lengths. So, trim out the `\0` replacing it with a blank to get blank padded `buf`.

Then, the old code on lines L7989 and L8000 will ensure that we will print just the leading 24-bytes.

These generated strings are thereby blank-extended in the output to get the desired well-aligned output.

Reply:
Use printf's field width instead.

Reply:
(robj) On L 7988 below, it seems like if you change the print fmt specifier to `%*.*s`, and pass 24 twice to specify min and field-width ... it may work.

If so, the jugglery about could all go away.

Reply:
Thanks, @rtjohnso - for this tip. Now, using `%*.*s` as suggested, prints the field's output correctly aligned.

Got rid of some crufty code.